src/errors.rs

```diff
@@ -0,0 +1,123 @@
+use crate::{
+    protos::coresdk::activity_result::ActivityResult,
+    protos::coresdk::workflow_completion::WfActivationCompletion,
+    protos::temporal::api::workflowservice::v1::PollWorkflowTaskQueueResponse,
+    workflow::WorkflowError,
+};
+use tonic::codegen::http::uri::InvalidUri;
+
+pub(crate) struct ShutdownErr;
+pub(crate) struct WorkflowUpdateError {
+    /// Underlying workflow error
+    pub source: WorkflowError,
+    /// The run id of the erring workflow
+    pub run_id: String,
+}
+
+/// Errors thrown during initialization of [crate::Core]
+#[derive(thiserror::Error, Debug, displaydoc::Display)]
+pub enum CoreInitError {
```

Comment:
I'd like us to have clearer instructions for lang for what needs to happen in each of these cases.
e.g. `CoreInitError` -> `InvalidUri` is fatal, we should create a new core, `TonicTransportError`, when is this retry-able who should retry (lang / core)?

Reply:
Initialization errors might as well always be considered fatal. It's so early on that just throwing everything out and trying again seems easiest and simplest